src/info/manifest.c

```diff
@@ -1552,6 +2322,55 @@ manifestFileTotal(const Manifest *this)
     FUNCTION_TEST_RETURN(lstSize(this->fileList));
 }
 
+void
+manifestFileUpdate(
+    Manifest *this, const String *name, uint64_t size, uint64_t sizeRepo, const char *checksumSha1, const String *reference,
+    bool checksumPage, bool checksumPageError, const VariantList *checksumPageErrorList)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(MANIFEST, this);
+        FUNCTION_TEST_PARAM(STRING, name);
+        FUNCTION_TEST_PARAM(UINT64, size);
+        FUNCTION_TEST_PARAM(UINT64, sizeRepo);
+        FUNCTION_TEST_PARAM(STRINGZ, checksumSha1);
+        FUNCTION_TEST_PARAM(STRING, reference);
+        FUNCTION_TEST_PARAM(BOOL, checksumPage);
+        FUNCTION_TEST_PARAM(BOOL, checksumPageError);
+        FUNCTION_TEST_PARAM(VARIANT_LIST, checksumPageErrorList);
+    FUNCTION_TEST_END();
+
+    ASSERT(this != NULL);
+    ASSERT(name != NULL);
+    ASSERT(
+        (!checksumPage && !checksumPageError && checksumPageErrorList == NULL) ||
+        (checksumPage && !checksumPageError && checksumPageErrorList == NULL) || (checksumPage && checksumPageError));
+
+    ManifestFile *file = (ManifestFile *)manifestFileFind(this, name);
+
+    MEM_CONTEXT_BEGIN(lstMemContext(this->fileList))
+    {
+        // Update reference if set
+        if (reference != NULL)
+            file->reference = strLstAddIfMissing(this->referenceList, reference);
+
+        // Update checksum if set
+        if (checksumSha1 != NULL)
+            memcpy(file->checksumSha1, checksumSha1, HASH_TYPE_SHA1_SIZE_HEX + 1);
+
+        // Update repo size
+        file->size = size;
+        file->sizeRepo = sizeRepo;
```

Comment:
The file may not have a repo-size (see below) so what are we really passing to this function if it doesn't? zero?:

pg_data/backup_label={"checksum":"0aaa9318dc6a0d461f28f5e649c8f9e18a623220","master":true,"size":229,"timestamp":1574780855}

Reply:
The file always has a repo size, we just don't store it if it is the same as size to save space.